File: ruby/6kyu/multiples-of-3-or-5.rb
# https://www.codewars.com/kata/514b92a657cdc65150000006/ruby

# --------------my solution--------------
def solution(number)
    sum = 0;
    
    for i in 0..(number-1)
      if i % 5 == 0 || i % 3 == 0 then
        sum += i;
      end  
    end
      
    return sum;
end

# --------------best practice--------------
def solution(number)
    (1...number).select {|i| i%3==0 || i%5==0}.inject(:+)
end

File: c#/6kyu/multiples-of-3-or-5.cs
// https://www.codewars.com/kata/514b92a657cdc65150000006/csharp

// ----------------my solution----------------
public static class Kata
{
  public static int Solution(int value)
  {
    int sum = 0;
    
    for (int i = 0; i < value; i++) {
      if (i % 3 == 0 || i % 5 == 0) sum += i;
    }
    
    return sum;
  }
}

// ----------------best practice----------------
//using System.Linq;
public static class Kata
{
    public static int Solution(int n)
    {
        return Enumerable.Range(0, n).Where(e => e % 3 == 0 || e % 5 == 0).Sum();
    }
}